guides/source/association_basics.md

```diff
@@ -2724,3 +2724,156 @@ will run a query like:
 ```sql
 SELECT "vehicles".* FROM "vehicles" WHERE "vehicles"."type" IN ('Car')
 ```
+
+Delegated Types
+----------------
+
+[`Single Table Inheritance (STI)`](#single-table-inheritance-sti) works best when there is little difference between subclasses and their attributes, but includes all attributes of all subclasses I need to create a single table.
+
+This has the disadvantage that attributes that are only used by a specific subclass are added to the same table, even though they are unnecessary attributes for another subclass, which bloats a single table.
+
+In the following example, the `Comment` does not need a `subject`, but the `Message` need a `subject`, so the `Entry` includes the `subject` attribute.
+
+```ruby
+# Schema: entries[ id, type, subject, created_at, updated_at]
+class Entry < ApplicationRecord
+end
+
+class Comment < Entry
+end
+
+class Message < Entry
+end
+```
+
+`Delegated types` solves this problem.
+
+A superclass has a table that stores attributes that are shared among all subclasses.
+Each subclass should have a separate table for additional attributes specific to its implementation.
+This eliminates the need to define attributes in one table that are only used by a particular subclass.
+
+First, let's generate the base `Entry` model:
+
+```bash
+$ bin/rails generate model entry entryable_type:string entryable_id:integer
+```
+
+The generated model will look like this:
+
+```ruby
+# Schema: entries[ id, entryable_type, entryable_id, created_at, updated_at ]
+class Entry < ApplicationRecord
+end
+```
+
+Then, generate the `Message` and `Comment` models for the delegation:
+
+```bash
+$ bin/rails generate model message subject:string body:string
+$ bin/rails generate model comment content:string
+```
+
+The generated model will look like this:
+
+```ruby
+# Schema: messages[ id, subject, body, created_at, updated_at ]
+class Message < ApplicationRecord
+end
+
+# Schema: comments[ id, content, created_at, updated_at ]
+class Comment < ApplicationRecord
+end
+```
+
+### Declare `delegated_type`
+
+First, declare a `delegated_type` in the superclass `Entry`.
+
+```ruby
+# Schema: entries[ id, entryable_type, entryable_id, created_at, updated_at ]
```

Comment:
```suggestion
```

I think we don't need to repeat the schema every time.

Reply:
OK, i have removed these.